crates/contract/tests/sandbox/integration_update_votes_cleanup_after_resharing.rs

```diff
@@ -0,0 +1,191 @@
+use anyhow::Result;
+use contract_interface::types as dtos;
+use near_account_id::AccountId;
+use serde_json::json;
+use std::collections::HashSet;
+use utilities::AccountIdExtV1;
+
+use crate::sandbox::common::{
+    assert_running_return_participants, assert_running_return_threshold,
+    execute_async_transactions, init_env, GAS_FOR_VOTE_RESHARED, PARTICIPANT_LEN,
+};
+use mpc_contract::{
+    primitives::{
+        domain::{DomainId, SignatureScheme},
+        key_state::EpochId,
+        participants::Participants,
+        thresholds::ThresholdParameters,
+    },
+    state::ProtocolContractState,
+    update::{ProposeUpdateArgs, UpdateId},
+};
+
+/// Tests that update votes from non-participants are cleared after resharing.
+#[tokio::test]
+async fn test_update_votes_cleared_after_resharing() -> Result<()> {
+    let (_, contract, env_accounts, _) =
+        init_env(&[SignatureScheme::Secp256k1], PARTICIPANT_LEN).await;
+
+    let initial_participants = assert_running_return_participants(&contract).await?;
+    let threshold = assert_running_return_threshold(&contract).await;
+
+    // Propose update and have first 2 participants vote on it
+    let update_id: UpdateId = env_accounts[0]
+        .call(contract.id(), "propose_update")
+        .args_borsh(ProposeUpdateArgs {
+            code: Some(vec![1u8; 1000]),
+            config: None,
+        })
+        .deposit(near_workspaces::types::NearToken::from_near(1))
+        .max_gas()
+        .transact()
+        .await?
+        .json()?;
+
+    for account in &env_accounts[0..2] {
+        account
+            .call(contract.id(), "vote_update")
+            .args_json(json!({"id": update_id}))
+            .max_gas()
+            .transact()
+            .await?
+            .into_result()?;
+    }
```

Comment:
for this we have a helper function now, which is probably better to use here: `execute_async_transactions`

Reply:
https://github.com/near/mpc/pull/1559/commits/b95755431435d14aae517856976f945c7e6b84dd